### PR TITLE
Feature/DGI-45: Command to update user

### PR DESCRIPTION
**Pull request by ntw-ITM**
Issue DGI-45 was originally about making a command to update specifically an accountant, but as I see it, it might as well be more general for updating a user. I assume that the account or role of the user cannot be updated, and that if an accountant should be removed from an account, it should be deactivated.

One could argue, that if the user to update is a 'client' (not an accountant), and the user is updated to a deactivated state, the account should also be deactivated, as well as any accountant on that account.

### CODE REVIEW

File: backend/Application/Users/Commands/UpdateUserCommand/UpdateUserCommand.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        if (_context.Accounts.Any(e => e.Email == request.User.Email) || _context.Users.Any(e => e.Email == request.User.Email && e.Id != request.User.Id))
+        {
+          throw new ArgumentException("The provided email address is already used by another user.");
+        }
+
+        userEntity.Name = request.User.Name ?? userEntity.Name;
+        userEntity.Email = request.User.Email ?? userEntity.Email;
+        userEntity.DeactivationTime = request.User.DeactivationTime ?? userEntity.DeactivationTime;
```
Comment (objmj-itminds) [inline backend/Application/Users/Commands/UpdateUserCommand/UpdateUserCommand.cs:47]: I'm not too fond of the deactivation time being part of the update here. I would suggest making another command that takes an ID and then sets the deactivation time from the backend side. This way, we avoid issues with timezones and MIM issues,

Comment (ntw-ITM) [reply backend/Application/Users/Commands/UpdateUserCommand/UpdateUserCommand.cs]: That makes sense. I now created another command just to deactivate a user. I also updated the UserController with endpoints to these commands.

File: backend/Application.UnitTests/Application.UnitTests.csproj [hunk 2]
```diff
@@ -26,4 +26,8 @@
     <ProjectReference Include="..\Infrastructure\Infrastructure.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Users\Commands\DeactivateUserCommandTest\" />
+  </ItemGroup>
+
```
Comment (objmj-itminds) [inline backend/Application.UnitTests/Application.UnitTests.csproj:32]: I don't know why you are generating these files or adding them to the commit. But please, don't.

File: backend/Web/Controllers/UserController.cs [hunk 3]
```diff
@@ -1,27 +1,49 @@
...
+    [HttpPut("{id}")]
+    public async Task<ActionResult> UpdateUser([FromRoute] int id, UpdateUserCommand command)
+    {
+      command.Id = id;
+      await Mediator.Send(command);
+
+      return NoContent();
+    }
+
+    [HttpPut("deactivate/{id}")]
```
Comment (objmj-itminds) [inline backend/Web/Controllers/UserController.cs:30]: "{id}/deactivate"

File: backend/Web/Controllers/UserController.cs [hunk 4]
```diff
@@ -1,27 +1,49 @@
...
+    {
+      await Mediator.Send(new DeactivateUserCommand
+      {
+        Id = id
+      });
+
+      return NoContent();
+    }
+
+    [HttpPut("changePassword/{id}")]
```
Comment (objmj-itminds) [inline backend/Web/Controllers/UserController.cs:41]: "{id}/changePassword"

**Review by objmj-itminds — CHANGES_REQUESTED** [review]

File: backend/Application.UnitTests/Application.UnitTests.csproj (hunk 2 above)
Comment (ntw-ITM) [reply backend/Application.UnitTests/Application.UnitTests.csproj:32]: Visual studio generates these changes whenever I add a new folder. I unstage them all the time, and I usually check all file changes before a PR, but I guess I missed this one.

**Comment by ntw-ITM** [comment]
Added requested changes.

**Review by objmj-itminds — APPROVED** [review]

**Pull request merged**